Move Concordion.NET to .NET Standard and NUnit v3

KidFashion opened the pull request:
This is a first step at moving legacy concordion.net baseline to .NET Standard and NUnitv3.

Currently code compiles and tests [succeed](https://ci.appveyor.com/project/KidFashion/concordion-net/build/1.0.28/tests) but we need to rebuild NUnit integration because NUnitv2 AddIn has been completely replaced in NUnitv3 [as explained here](https://github.com/nunit/docs/wiki/Addin-Replacement-in-the-Framework).

Commit:
Moved csproj to Framework 4.6.1

Commit:
Added version compiling targetting .NET Standard 2.0

Commit:
Added version compiling targetting .NET Standard 2.0

Commit:
Merge branch 'dev-build' of https://github.com/KidFashion/concordion-net into dev-build

Commit:
Added net 4.6.1 target in order to complete tests

Commit:
Added conditional reference for System.Drawing library

Commit:
Spec tests should pass even if BaseInputDirectory still needs fixes

Commit:
Spec tests should pass even if BaseInputDirectory still needs fixes

Commit:
Merge branch 'dev-build' of https://github.com/KidFashion/concordion-net into dev-build

Commit:
Merge branch 'dev-build' of https://github.com/KidFashion/concordion-net into dev-build

Commit:
Merge branch 'dev-build' of https://github.com/KidFashion/concordion-net into dev-build

Commit:
Merge branch 'dev-build' of https://github.com/KidFashion/concordion-net into dev-build

Commit:
Merge branch 'dev-build' of https://github.com/KidFashion/concordion-net into dev-build

Commit:
Fixed csproj for compiling also 4.6.1

Commit:
initial commit of NUnit 3 lib

Commit:
aggiunto pacchetto locale temporaneo per dll nunit.engine.api

ShaKaRee commented:
Hello Angelo,

thank you very much for contributing to the Concordion.NET project. Your help is highly welcome.

Before, we include your changes to the project, could we clean it up a bit?

- Could you remove "CAZZONE" from the source code?
- Could you remove all debugging code such as lines with Console.WriteLine?

Cheers

Commit:
cleaned up a bit code

KidFashion commented:
Hello ShaKaRee,

Thank you for spotting those issues, I've now cleaned up a bit, keep in mind two things:

- I've removed Console.WriteLines added by me (that were already commented out) but kept Console.WriteLines already present in code before (if that's a problem just make me know).
- This version is not intended to be released as it is, as discussed in Concordion Slack I worked on it this summer and now I've problems in finding more time to work on it so it was suggested me to open a pull request in order to start sharing work with others. If you prefer to add it to a nunit3 branch or whatever instead of targeting master I'm more than happy to redirect the PR there.

ShaKaRee commented:
Hello Angelo,

great that you share your work as soon as possible with others. I would like to include it directly into the master branch and I am looking forward to see the NUnit3 integration growing. :-)

Thank you for cleaning up the code.

Could you do a final step before we include it to the main branch? Could you remove the .old files? I think git provides a great history so that we can recover the old files easily.

Cheers

nigelcharman commented:
Hi, I'm in 2 minds about pulling to master:

1. master should always be in a releasable state. My understanding is that this code breaks the NUnit integration.
2. The concordion-net project isn't live. This is a step forward that takes us towards a working version of NUnit 3 code.

Keen to understand your thoughts on merging to master, @KidFashion, @ShaKaRee?

nigelcharman commented:
Also, should we set up CI for this branch? Using AppVeyor or something else?

nigelcharman commented:
Finally, can we clearly comment on the concordion.net and concordion-net projects, so users can understand the status of each project?

Commit:
removed .old files from repositories

KidFashion commented:
@ShaKaRee: I've removed the .old files.
@nigelcharman: Current code does not break NUnit integration, I started by making the current version compilable on .NET Core (this means with a working NUnit2 integration) and then I created a new Concordion.NUnit3 project where I started working on NUnit3 integration.

This means that we're not breaking NUnit2 but you will have an half finished (and not working) version of NUnit3 support, this is why I was asking if it was better to target a different branch from master. 

I think we should have a CI for each branch and AppVeyor works fine for me.

ShaKaRee commented:
Hi guys,
what about mergin the start of NUnit 3 integration into the main project?
@KidFashion perhaps you will find some time in near future to finish it. :-)
Cheers

KidFashion commented:
@ShaKaRee Indeed, I started it during summer holidays, so Christmas holidays seems a nice period to complete it :)

nigelcharman commented:
@KidFashion any chance of an update on this PR? I've mentioned it in the discussion about future .NET direction at concordion/concordion.net#15.

KidFashion commented:
> @KidFashion any chance of an update on this PR? I've mentioned it in the discussion about future .NET direction at [concordion/concordion.net#15](https://github.com/concordion/concordion.net/issues/15).

Unfortunately I've still to find time to look into that, since you pinged me I could look into that in the next two weeks but can't promise... :(

jamiehankins commented:
There are a few bugs relating to PRE tags that are causing us issues.
1. The parser gobbles up spaces between SPAN tags if space is all there is.
2. If you use an "&NBSP;", the parser throws an exception. Instead, you have to use "`&#160;`" (ugly!)
3. The parser includes spaces at the beginning of the PRE object for indenting, even though PRE content is supposed to be left alone.

I'm guessing that fixing any of this is out of the question with this PR hanging over things.

nigelcharman commented:
@jamiehankins as you might have guessed this PR has stalled. 

Would you raise your bugs as issues please, with a test case to show them? I'd like to see how the Java version of Concordion behaves with the same spec.

I suggest we then reach out to our users to see if anyone has the bandwidth to fix them.

jamiehankins commented:
@nigelcharman done as issues #28 #29 and #30

nigelcharman commented:
Thanks @jamiehankins , can you confirm which code base you are using? Is it `concordion.net` or `concordion-net`? If the latter, have you applied this PR? Just so we can be clear to anyone wanting to contribute fixes. Thanks

jamiehankins commented:
@nigelcharman Oops, I missed your last message. I'm no longer with the company I was working for, so I don't know the answer to your main question. The answer to the second question is no, we never applied the PR.

I have a job offer that I'm likely to accept later today. The job involves a lot of C#, so I might evangelize Concordion.NET. If that works out, maybe I can get buy-in to work on this.
